### PR TITLE
Adding 'no-floating-promises' lint rule, and fixing more floating pro…

### PR DESCRIPTION
**Pull request by EricCornelson**
…mises.

### CODE REVIEW

**Comment by roblourens** [comment]
@digeff I'll wait for you to review

File: src/chrome/internal/sources/features/notifyClientOfLoadedSources.ts [hunk 1]
```diff
@@ -35,13 +35,15 @@ export class NotifyClientOfLoadedSources {
     /**
      * e.g. the target navigated
      */
-    protected onExecutionContextsCleared(): void {
+    protected async onExecutionContextsCleared() {
```
Comment (digeff) [inline src/chrome/internal/sources/features/notifyClientOfLoadedSources.ts]: Return type Promise<void> ?

File: src/chrome/internal/sources/features/notifyClientOfLoadedSources.ts [hunk 2]
```diff
@@ -35,13 +35,15 @@ export class NotifyClientOfLoadedSources {
...
+        let sourceEvents = [];
         for (const script of this._notifiedSourceByIdentifier.values()) {
-            this.sendLoadedSourceEvent(script, 'removed');
+            sourceEvents.push(this.sendLoadedSourceEvent(script, 'removed'));
         }
+        return Promise.all(sourceEvents);
     }
 
-    protected sendLoadedSourceEvent(source: ILoadedSource, loadedSourceEventReason: LoadedSourceEventReason): void {
+    protected async sendLoadedSourceEvent(source: ILoadedSource, loadedSourceEventReason: LoadedSourceEventReason) {
```
Comment (digeff) [inline src/chrome/internal/sources/features/notifyClientOfLoadedSources.ts]: Return type Promise< void > ?

File: src/chrome/internal/sources/features/notifyClientOfLoadedSources.ts (hunk 1 above)
Comment (EricCornelson) [reply src/chrome/internal/sources/features/notifyClientOfLoadedSources.ts]: Fixed

File: src/chrome/internal/sources/features/notifyClientOfLoadedSources.ts (hunk 2 above)
Comment (EricCornelson) [reply src/chrome/internal/sources/features/notifyClientOfLoadedSources.ts]: Fixed

**Pull request merged**